library/core/src/ptr/non_null.rs

```diff
@@ -155,6 +165,18 @@ impl<T: ?Sized> NonNull<T> {
     /// # Safety
     ///
     /// `ptr` must be non-null.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::ptr::NonNull;
+    ///
+    /// let mut x = 0u32;
+    /// let ptr = unsafe { NonNull::new_unchecked(&mut x as *mut _) };
+    ///
+    /// // NEVER DO THAT!!!
+    /// let ptr = unsafe { NonNull::<u32>::new_unchecked(std::ptr::null_mut()) };
```

Comment:
Please mark testcases that cause UB as `no_run` (see e.g. the `MaybeUninit` docs).
Otherwise, https://github.com/rust-lang/miri-test-libstd/ will fail.

Reply:
See https://github.com/rust-lang/rust/pull/87653

Reply:
Noted!